components/buildless-serverless/internal/controller/validator/validator.go

```diff
@@ -16,11 +20,13 @@ import (
 
 type validator struct {
 	instance *serverlessv1alpha2.Function
+	fnConfig config.FunctionConfig
 }
 
-func New(instance *serverlessv1alpha2.Function) *validator {
+func New(m *fsm.StateMachine) *validator {
```

Comment:
Now `Validator` depends on `StateMachine`. Could we pass only function and function config?

Reply:
Thanks, adjusted to pass 2 vars instead of whole `StateMachine`